File: src/main/java/com/pelican/Material.java
package com.pelican;

import org.lwjgl.assimp.AIColor4D;
import org.lwjgl.assimp.AIMaterial;

import static org.lwjgl.assimp.Assimp.*;

/**
 * @author Ryan Benasutti
 * @since 05-24-2017
 */
public class Material {
    private AIMaterial mat;
    private AIColor4D ambient, diffuse, specular;

    public Material(AIMaterial material) {
        this.mat = material;

        ambient = AIColor4D.create();
        if (aiGetMaterialColor(mat, AI_MATKEY_COLOR_AMBIENT, aiTextureType_NONE, 0, ambient) != 0) {
            throw new IllegalStateException(aiGetErrorString());
        }

        diffuse = AIColor4D.create();
        if (aiGetMaterialColor(mat, AI_MATKEY_COLOR_DIFFUSE, aiTextureType_NONE, 0, diffuse) != 0) {
            throw new IllegalStateException(aiGetErrorString());
        }

        specular = AIColor4D.create();
        if (aiGetMaterialColor(mat, AI_MATKEY_COLOR_SPECULAR, aiTextureType_NONE, 0, specular) != 0) {
            throw new IllegalStateException(aiGetErrorString());
        }
    }
}

File: src/main/java/com/pelican/Material.kt
package com.pelican

import org.lwjgl.assimp.AIColor4D
import org.lwjgl.assimp.AIMaterial
import org.lwjgl.assimp.Assimp.*

/**
 * @author Ryan Benasutti
 * @since 05-24-2017
 * License terms: https://www.github.com/Octogonapus/Pelican/blob/master/LICENSE.md
 */
class Material(mMaterial: AIMaterial) {
    var mAmbientColor: AIColor4D
    var mDiffuseColor: AIColor4D
    var mSpecularColor: AIColor4D

    init {

        val numProps = mMaterial.mNumProperties()
        val buf = mMaterial.mProperties()
        for (i in 0 until numProps) {
            println(buf.getStringASCII(i))
        }

        mAmbientColor = AIColor4D.create()
        if (aiGetMaterialColor(mMaterial, AI_MATKEY_COLOR_AMBIENT,
                aiTextureType_NONE, 0, mAmbientColor) != 0) {
            throw IllegalStateException(aiGetErrorString())
        }
        mDiffuseColor = AIColor4D.create()
        if (aiGetMaterialColor(mMaterial, AI_MATKEY_COLOR_DIFFUSE,
                aiTextureType_NONE, 0, mDiffuseColor) != 0) {
            throw IllegalStateException(aiGetErrorString())
        }
        mSpecularColor = AIColor4D.create()
        if (aiGetMaterialColor(mMaterial, AI_MATKEY_COLOR_SPECULAR,
                aiTextureType_NONE, 0, mSpecularColor) != 0) {
            throw IllegalStateException(aiGetErrorString())
        }
    }

}
